test/unit/org/apache/cassandra/index/sai/cql/AllowFilteringTest.java

```diff
@@ -296,6 +296,79 @@ public void testAllowFilteringOnClusteringAndRegularColumns() throws Throwable
         test("SELECT * FROM %s WHERE v1=0 AND v2=0 AND k1=0 AND k2=0 AND (c1, c2, c3, c4) = (0, 0, 0, 0) AND v3=0", true);
     }
 
+    @Test
+    public void testAllowFilteringTextWithINClause () throws Throwable
+    {
+        createTable("CREATE TABLE %S (K1 TEXT, K2 TEXT, K3 TEXT, PRIMARY KEY(K1))");
+        createIndex(format("CREATE CUSTOM INDEX ON %%s(K2) USING '%s'", StorageAttachedIndex.class.getName()));
```

Comment:
nit: In new code, I would start using the simplified version:

```suggestion
    createIndex("CREATE INDEX ON %s(K2) USING 'sai'");
```

Reply:
Don't have to change everything else in this test, but I'd be fine if you wanted to...